### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by amexabee**
### Implement classes and objects in Ruby.
- Create Person class
- Create Student class 
- Create Teacher class
### Implement encapsulation and inheritance with Ruby.
- make use of getters and setters
- Student and Teacher class extend Person class

### CODE REVIEW

File: src/person.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def can_use_services?
+    return true if of_age? || @parent_permission
+
+    false
```
Comment (Deepakdanger) [inline src/person.rb]: - [ ] [OPTIONAL] No need to write the true or false, You can use it directly `of_age?() || @parent_permission`.

File: src/person.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
```
Comment (Deepakdanger) [inline src/person.rb]: - [ ] [OPTIONAL] No need to write the true or false, You can use it directly 
```
def of_age?
    @age >= 18
  end
```

**Review by Deepakdanger — APPROVED** [review]
Hi @amexabee , 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- _Check the comment below._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/person.rb (hunk 1 above)
Comment (amexabee) [reply src/person.rb]: thank you!

File: src/person.rb (hunk 2 above)
Comment (amexabee) [reply src/person.rb]: thank you!

**Pull request merged**